DocumentDataAPI/DocumentDataAPI/Data/Repositories/NpgSimilarDocumentRepository.cs

```diff
@@ -0,0 +1,92 @@
+        _logger = logger;
+        _sqlHelper = sqlHelper;
+    }
+
+    public async Task<long> Add(SimilarDocumentModel entity)
+    {
+        _logger.LogDebug("Adding Source with mainDocumentId: {mainDocumentId} and" +
+            " similarDocumentId: {similarDocumentId} to database", entity.MainDocumentId, entity.SimilarDocumentId);
+        _logger.LogTrace("similarDocument: {similarDocument}", entity);
+        using IDbConnection con = _connectionFactory.CreateConnection();
+        return await con.QuerySingleAsync<long>($"insert into similar_documents({SimilarDocumentMap.MainDocumentId}, {SimilarDocumentMap.SimilarDocumentId}) " +
+            $"values (@MainDocumentId, @SimilarDocumentId) returning {SimilarDocumentMap.MainDocumentId}",
+            new { entity.MainDocumentId, entity.SimilarDocumentId });
+    }
+
+    public async Task<IEnumerable<long>> AddBatch(List<SimilarDocumentModel> models)
+    {
+        IEnumerable<long> allInsertedIds = new List<long>();
+        _logger.LogDebug("Adding {count} SimilarDocument to database", models.Count);
+        using IDbConnection con = _connectionFactory.CreateConnection();
+        con.Open();
+        using IDbTransaction transaction = con.BeginTransaction();
+        try
+        {
+            // Divide the list of models into chunks to keep the INSERT statements from getting too large.
+            foreach (SimilarDocumentModel[] chunk in models.Chunk(_sqlHelper.InsertStatementChunkSize))
+            {
+                string parameterString = _sqlHelper.GetBatchInsertParameters(chunk, out Dictionary<string, dynamic> parameters);
+                IEnumerable<long> insertedIds = await transaction.QueryAsync<long>(
+                        $"insert into similar_documents({SimilarDocumentMap.MainDocumentId}, {SimilarDocumentMap.SimilarDocumentId}, {SimilarDocumentMap.Similarity}) " +
+                        $"values {parameterString} returning {SimilarDocumentMap.MainDocumentId}",
+                    parameters);
+                allInsertedIds = allInsertedIds.Concat(insertedIds);
+            }
+            transaction.Commit();
+        }
+        catch (Exception e)
+        {
+            transaction.Rollback();
+            _logger.LogError(e, "Failed to insert word ratios, rolling back transaction");
+            throw;
+        }
+        return allInsertedIds;
+    }
+
+    public async Task<int> Delete(long mainDocumentId, long similarDocumentId)
+    {
+        _logger.LogDebug("Deleting Document with id {mainDocumentId} {similarDocumentId} from database", mainDocumentId, similarDocumentId);
+        using IDbConnection con = _connectionFactory.CreateConnection();
+        return await con.ExecuteAsync($"delete from similar_documents where {SimilarDocumentMap.MainDocumentId} = @mainDocumentId and " +
+            $"{SimilarDocumentMap.SimilarDocumentId} = @similarDocumentId",
+            new { mainDocumentId = mainDocumentId, similarDocumentId = similarDocumentId });
```

Comment:
```suggestion
            new { mainDocumentId, similarDocumentId });
```
Det burde ikke være nødvendigt at gøre det andet